datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -137,8 +137,15 @@ impl TableProviderFactory for ListingTableFactory {
         let mut statement_options = StatementOptions::from(&cmd.options);
 
         // Extract ListingTable specific options if present or set default
-        // Discard unbounded option if present
-        statement_options.take_str_option("unbounded");
+        let unbounded = if infinite_source {
```

Comment:
You can remove this logic since we support unbounded tables with `CREATE UNBOUNDED EXTERNAL TABLE`.

Reply:
That syntax is supported, but the test for FIFO insert passes unbounded as a string option. The logic here was added to handle either case. Create unbounded syntax takes precedence, but if it is not present we check for OPTION (unbounded true). 

https://github.com/apache/arrow-datafusion/blob/1dd0e988ceb6ecc1df4053fd7d065c15c6e8b431/datafusion/core/tests/fifo.rs#L399

Reply:
I see, I think it is a leftover. I guess we can change it into Create Unbounded External if I am not missing anything.

Reply:
I think it is OK if both methods are supported. That would provide backwards compatibility for anyone who used this syntax in past versions.